Lib/test/test_tools/test_makefile.py

```diff
@@ -0,0 +1,66 @@
+"""
+Tests for `Makefile`.
+"""
+
+import os
+import unittest
+from test import support
+
+MAKEFILE = os.path.join(support.REPO_ROOT, 'Makefile')
+
+if not support.check_impl_detail(cpython=True):
+    raise unittest.SkipTest('cpython only')
+if support.MS_WINDOWS:
+    raise unittest.SkipTest('Windows does not support make')
+if not os.path.exists(MAKEFILE) or not os.path.isfile(MAKEFILE):
+    raise unittest.SkipTest('Makefile could not be found')
+
+try:
+    import _testcapi
+except ImportError:
+    TEST_MODULES = False
+else:
+    TEST_MODULES = True
+
+@support.requires_subprocess()
+class TestMakefile(unittest.TestCase):
+    def list_test_dirs(self):
+        import subprocess
+        return subprocess.check_output(
+            ['make', '-s', '-C', support.REPO_ROOT, 'listtestmodules'],
+            universal_newlines=True,
+            encoding='utf-8',
+        )
+
+    def check_existing_test_modules(self, result):
+        test_dirs = result.split(' ')
+        idle_test = 'idlelib/idle_test'
+        self.assertIn(idle_test, test_dirs)
+
+        used = [idle_test]
+        for dirpath, _, _ in os.walk(support.TEST_HOME_DIR):
+            dirname = os.path.basename(dirpath)
+            if dirname == '__pycache__':
+                continue
+            with self.subTest(dirpath=dirpath):
+                relpath = os.path.relpath(dirpath, support.STDLIB_DIR)
+                self.assertIn(relpath, test_dirs)
```

Comment:
The error message if this assertion fails is very verbose but not very helpful: https://github.com/python/cpython/actions/runs/8004529977/job/21862119274?pr=115813

Maybe something like this?

```suggestion
                self.assertIn(
                    relpath,
                    test_dirs,
                    msg=(
                        f"{relpath!r} is not included in the Makefile's list "
                        f"of test directories to install"
                    )
                )
```